rdtools/test/soiling_test.py

```diff
@@ -57,12 +57,34 @@ def test_soiling_srr(soiling_normalized_daily, soiling_insolation, soiling_times
     pd.testing.assert_index_equal(soiling_info['soiling_ratio_perfect_clean'].index, soiling_times,
                                   check_names=False)
     sr_mean = soiling_info['soiling_ratio_perfect_clean'].mean()
-    assert 0.967170 == pytest.approx(sr_mean, abs=1e-6),\
+    assert 0.968265 == pytest.approx(sr_mean, abs=1e-6),\
         "The mean of soiling_info['soiling_ratio_perfect_clean'] differs from expected"
     assert isinstance(soiling_info['soiling_ratio_perfect_clean'], pd.Series),\
         'soiling_info["soiling_ratio_perfect_clean"] not a pandas series'
 
 
+def test_soiling_srr_consecutive_invalid(soiling_normalized_daily, soiling_insolation,
+                                         soiling_times):
+    reps = 10
+    np.random.seed(1977)
+    sr, sr_ci, soiling_info = soiling_srr(soiling_normalized_daily, soiling_insolation, reps=reps,
+                                          max_relative_slope_error=20.0, method='random_clean')
+    assert 0.936177 == pytest.approx(sr, abs=1e-6),\
+        'Soiling ratio different from expected value for random_clean with consecutive invalid intervals'  # noqa: E501
+
+    np.random.seed(1977)
+    sr, sr_ci, soiling_info = soiling_srr(soiling_normalized_daily, soiling_insolation, reps=reps,
+                                          max_relative_slope_error=20.0, method='half_norm_clean')
+    assert 0.915093 == pytest.approx(sr, abs=1e-6),\
+        'Soiling ratio different from expected value for half_norm_clean with consecutive invalid intervals'  # noqa: E501
+
+    np.random.seed(1977)
+    sr, sr_ci, soiling_info = soiling_srr(soiling_normalized_daily, soiling_insolation, reps=reps,
+                                          max_relative_slope_error=20.0, method='perfect_clean')
+    assert 0.977116 == pytest.approx(sr, abs=1e-6),\
+        'Soiling ratio different from expected value for perfect_clean with consecutive invalid intervals'  # noqa: E501
```

Comment:
Maybe worth `parametrize`ing (though the same could be said of all the other tests too):
```python
@pytest.mark.parametrize('method,expected_sr',
                         [('random_clean', 0.936177),
                          ('half_norm_clean', 0.915093),
                          ('perfect_clean', 0.977116)])
def test_soiling_srr_consecutive_invalid(soiling_normalized_daily, soiling_insolation,
                                         soiling_times, method, expected_sr):
    reps = 10
    np.random.seed(1977)
    sr, sr_ci, soiling_info = soiling_srr(soiling_normalized_daily, soiling_insolation, reps=reps,
                                          max_relative_slope_error=20.0, method=method)
    assert expected_sr == pytest.approx(sr, abs=1e-6),\
        f'Soiling ratio different from expected value for {method} with consecutive invalid intervals'  # noqa: E501
```

Reply:
Good suggestion, I've done this to a few of the tests.